src/wowman/catalog.clj

```diff
@@ -39,15 +43,81 @@
      :total (count addon-list)
      :addon-summary-list addon-list}))
 
+(defn read-catalog
+  "read catalog data from given `catalog-path`"
+  [catalog-path & {:keys [bad-data?]}]
+  ;; todo: cheshire claims to be twice as fast: https://github.com/dakrone/cheshire#speed
+  (if bad-data?
```

Comment:
not great, as :bad-data? may be nil. can we just pass the args through without inspecting them?